### PR TITLE
Reduction scheduler cleanup

### PR DESCRIPTION
**Pull request by csarofeen**
Minor cleanup in the reduction scheduler. This mostly takes the approach in normalization and adapts it for reduction. It removes a variable I particularly didn't like the use of `min_red_elems_per_thread `.

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/compute_at.cpp [hunk 1]
```diff
@@ -699,11 +700,19 @@ void ComputeAt::updateSiblings() {
         }
         auto sibling_domain =
             TransformReplay::fullSelfReplay(sibling_tv->domain(), tv->domain());
-        sibling_tv->setDomain(sibling_domain);
-        sibling_tv->setComputeAt(tv->getComputeAtPosition());
-        sibling_tv->setMaxProducer(tv->getMaxProducerPosition());
+        if (tv->getComputeAtPosition() > sibling_tv->getComputeAtPosition()) {
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/compute_at.cpp:701]: nit: Can this check be done earlier and skip `fullSelfReplay` if it's false?

File: torch/csrc/jit/codegen/cuda/compute_at.cpp [hunk 2]
```diff
@@ -756,11 +765,12 @@ void ComputeAt::runPass() {
   // Back off on inlining the inner broadcast axes
   hoistInnermostBroadcast();
 
+  // Clear max producer position of consumers from fusion inputs.
+  updateInputProduceAts();
+
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/compute_at.cpp:770]: Why is `updateInputProduceAts` moved before `updateSiblings`? Not sure if there's any ordering requirement.

**Review by naoyam — APPROVED** [review]
LGTM. Just left a couple of minor questions.

File: torch/csrc/jit/codegen/cuda/compute_at.cpp (hunk 2 above)
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/compute_at.cpp:770]: I wanted to make sure all changes were done before updating the siblings. Just seemed to make more sense to me like that.

**Pull request merged**